StRoot/StarGenerator/FILT/FcsDYBGFilter.cxx

```diff
@@ -0,0 +1,120 @@
+#include "FcsDYBGFilter.h"
+
+#include "StarGenerator/EVENT/StarGenParticle.h"
+#include "StarGenerator/EVENT/StarGenEvent.h"
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <cmath>
+#include <vector>
+#include <algorithm>
+
+static const float ZFCS    = 710.16 + 13.9 + 15.0;
+static const float XFCSMin = 16.69;
+static const float XFCSMax = 16.69 + 22.0 * 5.542;
+static const float YFCS    = 34.0/2.0 * 5.542;
+static const float FVCUT   = 0;//2.5; //cm
+static const float ETCUT   = 0.8;
+static const float DRCUT   = 40.0;
+static const float DYETCUT = 1.0;
+static const float DYMASSCUT= 2.5;
+
+
+static int ntot=0;
+static int ngood=0;
+
+using namespace std;
+//_______________________________________________________________
+FcsDYBGFilter::FcsDYBGFilter():StarFilterMaker("fcsDYBGFilter") {
+    cout<<"FCS DYBG filter is used!!!"<<endl;
+    cout<<"FCS DYBG filter swap particles="<<mSwap<<endl;
+}
+
+FcsDYBGFilter::FcsDYBGFilter(int dy, int check, int swap):StarFilterMaker("fcsDYBGFilter"), mDYmode(dy), mCheckmode(check), mSwap(swap){
+    cout<<"FCS DYBG filter is used!!!"<<endl;
+    cout<<"FCS DYBG filter DYMode="<<mDYmode<<endl;
+    cout<<"FCS DYBG filter Checkmode="<<mCheckmode<<endl;
+    cout<<"FCS DYBG filter swap particles="<<mSwap<<endl;
+}
+//_______________________________________________________________
+Int_t FcsDYBGFilter::Filter( StarGenEvent *mEvent){
+    ntot++;
+    // Get a reference to the current event 
+    StarGenEvent& event = *mEvent;
+    
+    //event.Print();
+    int np=event.GetNumberOfParticles();
+    if(np <= 0) {return kError;}
```

Comment:
The filter method should only return an accept / reject decision.   If this line is commented out, the code will return a reject in the case where no particles were generated.  Recommend commenting this out.